### PR TITLE
Coin Change Problem Added

### PR DESCRIPTION
**Pull request by DEEPALI-SUK**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added the article for Coin Change Problem. The basic layout for the article is:
What is the Coin Change Problem?
Explanation for Coin Change Problem
Algorithmic Steps
Example
Code in Java and Python
Sample I/O
Time & Space Complexity

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Article for Coin Change Problem #126

### CODE REVIEW

**Comment by DEEPALI-SUK** [comment]
@HarshCasper Kindly, review my PR

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+#### Output
+
+```
+Minimum coins required are 2
+```
+
+## Complexity Analysis
+
+- `n` is the size of the coins array.
+- `amount` is the total required amount.
```
Comment (HarshCasper) [inline docs/dynamic-programming/coin-change-problem.mdx:238]: They don't need to be in a bullet point

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  # returns value in n * value block
+
+    return k[n][value]
+
+
+min_coins = coinChange(coins, value, n)
+
+# printing the min required coins
+
+print ('Minimum coins required are {}'.format(min_coins))
```
Comment (HarshCasper) [inline docs/dynamic-programming/coin-change-problem.mdx:211]: Put it in a main driver code

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 3]
```diff
@@ -0,0 +1,245 @@
+---
+title: Coin Change Problem
+slug: /dynamic-programming/coin-change-problem
+---
+
+## Introduction
+
+Given an integer `amount`, which represents the total amount of money, an integer array `coins` representing different coin values, and each denomination has an infinite supply of coins. Find what will be the minimum number of coins required to make up the given amount?
```
Comment (HarshCasper) [inline docs/dynamic-programming/coin-change-problem.mdx:8]: Can you give more context and information?

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+1. Make a 2-d array, say `dp[n+1][amount+1]`.
+2. Assign all the columns of the `0th` row of the 2d-array to a maximum value: `dp[0][i]= INT_MAX`.
+3. Assign all the rows of the `0th` column of the 2-d array as 0, that is, `dp[i][0]=0`.
+4. For the `1st` row, if the column indices value is divisible by the first coin, write the quotient in that cell, else write the max value.
+5. For row>1, iterate till column index value is less than the ith coin value (i.e for all column indices, j,such that, j<coins[i-1] )and assign `dp[i][j]` to the previous upper row value, that is, `dp[i][j]=dp[i-1][j]`.
+6. Now we have two options for all column indices, j, such that, j>=coins[i-1], whether to include ith coin or not.
+7. Case1- If we don't consider the ith coin, then the cell value is the previous row cell value that is, `dp[i][j]=dp[i-1][j]`.
+8. Case2- If we consider the ith coin, then the cell value is `1+ dp[i][j-coins[i-1]]` cell value.
+9. The final value of the cell is the minimum of both the cases,that is, `dp[i][j]= min(dp[i-1][j], 1+dp[i][j-coins[i-1]])`.
+10. The final answer will be `dp[n][amount]`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/coin-change-problem.mdx:25]: Use unordered list, not ordered ones

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+    *---*---*---*---*---*---*---*---*---*---*---*
+  2 | 0 | 1 | 2 | 3 | 4 | 1 | 2 | 3 | 4 | 5 | 2 |
+    *---*---*---*---*---*---*---*---*---*---*---*
+  3 | 0 | 1 | 2 | 3 | 4 | 1 | 1 | 2 | 3 | 4 | 2 |
+    *---*---*---*---*---*---*---*---*---*---*---*
+  4 | 0 | 1 | 2 | 3 | 4 | 1 | 1 | 2 | 3 | 1 | 2 |
+    *---*---*---*---*---*---*---*---*---*---*---*
+```
+
+`Answer = dp[4][10] = 2`
```
Comment (HarshCasper) [inline docs/dynamic-programming/coin-change-problem.mdx:53]: Better use a diagram here

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by DEEPALI-SUK** [comment]
@HarshCasper I am closing this PR and creating a new one with suggested changes as I am having problems in pulling. Sorry, for the inconvenience caused.

**Pull request closed without merging**